neuralpp/test/quick_tests/graphical_model/representation/factor/symbolic_factor_test.py

```diff
@@ -42,6 +68,30 @@ def test_mul_by_non_identity():
 
     assert symbolic1_condition.expression.sympy_object * symbolic2_condition.expression.sympy_object == symbolic3_condition.expression.sympy_object
 
+
+def test_if_then_else_mul_by_non_identity():
+    x = IntegerVariable("x", 3)
+    y = IntegerVariable("y", 2)
+    z = IntegerVariable("z", 3)
+
+    x_symbol, y_symbol, z_symbol = sympy.symbols("x y z")
+    x_sympy = SymPyVariable(x_symbol, int)
+    y_sympy = SymPyVariable(y_symbol, int)
+    z_sympy = SymPyVariable(z_symbol, int)
+
+    expression1 = if_then_else(x_sympy > y_sympy, 2, 3)
+    expression2 = if_then_else(x_sympy == z_sympy, 1, 5)
+
+    symbolic1 = SymbolicFactor([x, y], expression1)
+    symbolic2 = SymbolicFactor([x, z], expression2)
+    symbolic3 = symbolic1.mul_by_non_identity(symbolic2)
```

Comment:
I should have named `mul_by_non_identity` as a private method `_mul_by_non_identity`, because it is not supposed to be used externally. The operator `*` (`__mul__`) takes care of the identity case and calls `mul_by_non_identity` to deal with other cases, and that is the only place where this method should be called. If called elsewhere, like here, it may fail because it doesn't deal with the identity case necessarily.
In this case here you can simply write `symbolic3 = symbolic1 * symbolic2`.